src/components/transactions/SingleTx/index.tsx

```diff
@@ -1,59 +1,53 @@
-import { CircularProgress } from '@mui/material'
+import { Box, CircularProgress } from '@mui/material'
 import ErrorMessage from '@/components/tx/ErrorMessage'
-import useChainId from '@/hooks/useChainId'
-import { useRouter } from 'next/router'
 import useSafeInfo from '@/hooks/useSafeInfo'
-import useAsync from '@/hooks/useAsync'
-import type { DateLabel, Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
-import { getTransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
-import { sameAddress } from '@/utils/addresses'
+import type { Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
 import type { ReactElement } from 'react'
-import { makeDateLabelFromTx, makeTxFromDetails } from '@/utils/transactions'
+import { makeTxFromDetails } from '@/utils/transactions'
 import { TxListGrid } from '@/components/transactions/TxList'
-import TxDateLabel from '@/components/transactions/TxDateLabel'
 import ExpandableTransactionItem from '@/components/transactions/TxListItem/ExpandableTransactionItem'
+import { useTxDetails } from '@/hooks/useTxDetails'
```

Comment:
If this hook is used only here, I would put it in the same folder as this component.